From 4ee5b75067dcd616d93cd4e16221eb6e17c535d4 Mon Sep 17 00:00:00 2001
From: Chris Diamand <chris@diamand.org>
Date: Fri, 10 Apr 2015 17:55:00 +0100
Subject: [PATCH 01/11] relf.h: Make sure struct link_map is defined.

---
 include/relf.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/relf.h b/include/relf.h
index ddea7e6..ead8242 100644
--- a/include/relf.h
+++ b/include/relf.h
@@ -1,6 +1,7 @@
 #ifndef RELF_H_
 #define RELF_H_
 
+#include <link.h>
 #include <stddef.h> /* for offsetof */
 #include <elf.h>
 #include <string.h>
-- 
2.1.4

