From 7c3a599ee7c3fa67108a458ffe324398c86bc3f1 Mon Sep 17 00:00:00 2001
From: Chris Diamand <chris@diamand.org>
Date: Wed, 18 Mar 2015 23:57:20 +0000
Subject: [PATCH 06/11] Add some Makefile dependencies to allow parallel
 builds.

---
 Makefile       | 2 +-
 tools/Makefile | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 282dcf6..aca0451 100644
--- a/Makefile
+++ b/Makefile
@@ -28,5 +28,5 @@ clean:
 	$(MAKE) -C tests clean
 
 .PHONY: tests
-tests:
+tests: lib
 	$(MAKE) -C tests
diff --git a/tools/Makefile b/tools/Makefile
index 308d39d..86850de 100644
--- a/tools/Makefile
+++ b/tools/Makefile
@@ -10,7 +10,7 @@ CPP_SRC := $(wildcard *.cpp)
 default: liballocstool.a dumptypes allocsites usedtypes ifacetypes find-allocated-type-size lang
 
 .PHONY: lang
-lang:
+lang: liballocstool.a
 	for dir in $(wildcard lang/*); do if [ -e $$dir/Makefile ]; then $(MAKE) -C $$dir; fi; done
 
 .PHONY: clean
-- 
2.1.4

