From 3f515213a418a949542ab1f601bb05feb98b7f31 Mon Sep 17 00:00:00 2001
From: Chris Diamand <chris@diamand.org>
Date: Sun, 12 Apr 2015 00:03:42 +0100
Subject: [PATCH 09/11] Remove invalid assumption in stack frame sanity check.

---
 src/fake-libunwind.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/fake-libunwind.c b/src/fake-libunwind.c
index dff22c7..c2d35ad 100644
--- a/src/fake-libunwind.c
+++ b/src/fake-libunwind.c
@@ -109,8 +109,7 @@ int unw_step(unw_cursor_t *cp)
 	};
 		
 	// sanity check the results
-	if (new_ctxt.frame_sp >= BEGINNING_OF_STACK
-	||  new_ctxt.frame_sp <= (BEGINNING_OF_STACK - 0x100000000))
+	if (new_ctxt.frame_sp >= BEGINNING_OF_STACK)
 	{
 		// looks dodgy -- say we failed
 		return -1;
-- 
2.1.4

