From 27c26c0d01e2563278fa3f18a160a26d3de1dc10 Mon Sep 17 00:00:00 2001
From: Chris Diamand <chris@diamand.org>
Date: Sat, 28 Mar 2015 01:08:29 +0000
Subject: [PATCH 7/7] Don't assume location of dwarf.h.

---
 include/dwarfpp/Makefile | 10 ++++++++--
 src/Makefile             | 10 ++++++++--
 2 files changed, 16 insertions(+), 4 deletions(-)

diff --git a/include/dwarfpp/Makefile b/include/dwarfpp/Makefile
index 56dc24a..33ad7ed 100644
--- a/include/dwarfpp/Makefile
+++ b/include/dwarfpp/Makefile
@@ -12,12 +12,18 @@ default: $(patsubst %.m4,%,$(m4_hdrs)) dwarf-onlystd.h dwarf-onlystd-v2.h dwarf-
 clean:
 	rm -f $(hdrs)
 
-dwarf-onlystd.h: /usr/include/dwarf.h
+# Use CPP to search the include path looking for libdwarf.h (we can't just
+# assume it's in /usr/lib).
+DWARF_DOT_H := $(shell bash -c "echo '\#include <dwarf.h>' | \
+                                cpp -H -o /dev/null 2>&1 | \
+                                head -n1 | cut -d ' ' -f 2")
+
+dwarf-onlystd.h: $(DWARF_DOT_H)
 	cat "$<" | egrep -v 'DW_[A-Z]+_(GNU|SUN|HP|APPLE|INTEL|ARM|upc|PGI|ALTIUM|MIPS|CPQ|VMS|GNAT)' | \
 		egrep -v '/\* (SGI|GNU)( \*/|\. )' | egrep -v 'LANG_Mips|LANG_Upc' | egrep -v '_use_GNAT' | egrep -v 'ATCF entries start at|DW_LANG_UPC instead.' > "$@"
 
 dwarf-onlystd-v2.h: dwarf-onlystd.h
 	cat "$<" | grep -v 'DWARF[^2]' > "$@"
 
-dwarf-ext-GNU.h: /usr/include/dwarf.h
+dwarf-ext-GNU.h: $(DWARF_DOT_H)
 	cat "$<" | egrep '(_|/\* |, )GNU' | egrep -vi conflict | egrep -vi '^[[:blank:]]*/\*' > "$@"
diff --git a/src/Makefile b/src/Makefile
index 8ac85fd..5a72f04 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -40,14 +40,20 @@ $(DEPS): .%.d : %.cpp
 %.o: %.cpp #$(HDRS)
 	$(CXX) $(CXXFLAGS) -c -o "$@" "$<"
 
-dwarf-onlystd.h: /usr/include/dwarf.h
+# Use CPP to search the include path looking for libdwarf.h (we can't just
+# assume it's in /usr/lib).
+DWARF_DOT_H := $(shell bash -c "echo '\#include <dwarf.h>' | \
+                                cpp -H -o /dev/null 2>&1 | \
+                                head -n1 | cut -d ' ' -f 2")
+
+dwarf-onlystd.h: $(DWARF_DOT_H)
 	cat "$<" | egrep -v 'DW_[A-Z]+_(GNU|SUN|HP|APPLE|INTEL|ARM|upc|PGI|ALTIUM|MIPS|CPQ|VMS|GNAT)' | \
 		egrep -v '/\* (SGI|GNU)( \*/|\. )' | egrep -v 'LANG_Mips|LANG_Upc' | egrep -v '_use_GNAT' | egrep -v 'ATCF entries start at|DW_LANG_UPC instead.' > "$@"
 
 dwarf-onlystd-v2.h: dwarf-onlystd.h
 	cat "$<" | grep -v 'DWARF[^2]'
 
-dwarf-ext-GNU.h: /usr/include/dwarf.h
+dwarf-ext-GNU.h: $(DWARF_DOT_H)
 	cat "$<" | egrep '(_|/\* |, )GNU' | egrep -vi conflict | egrep -vi '^[[:blank:]]*/\*' > "$@"
 
 libdwarfpp.a: $(OBJS)
-- 
2.1.4

