From f10cae52dfa198889962b88737a84cffb3b02d5d Mon Sep 17 00:00:00 2001
From: Chris Diamand <chris@diamand.org>
Date: Sat, 27 Dec 2014 23:14:56 +0000
Subject: [PATCH 1/4] Add '.gitignore' file.

---
 .gitignore | 32 ++++++++++++++++++++++++++++++++
 1 file changed, 32 insertions(+)
 create mode 100644 .gitignore

diff --git a/.gitignore b/.gitignore
new file mode 100644
index 0000000..2102cd4
--- /dev/null
+++ b/.gitignore
@@ -0,0 +1,32 @@
+*.a
+*.allocstubs.c
+*.allocstubs.s
+*.cil.c
+*.cil.i
+*.cil.s
+*.cma
+*.cmi
+*.cmo
+*.cmx
+*.cmxs
+*.d
+*.i
+*.i.allocs
+*.o
+*.o.fixuplog
+*.pyc
+*.so
+*.swp
+tests/alloca/alloca
+tests/offsetof/offsetof
+tests/section-group/section-group
+tests/simple-client/simple-client
+tests/simple-multi-alloc/simple-multi-alloc
+tests/sloppy-dumptypes/sloppy-dumptypes
+tools/allocsites
+tools/dumptypes
+tools/ifacetypes
+tools/find-allocated-type-size
+tools/lang/c/bin
+tools/lang/c/src/base-types-translation
+tools/usedtypes
-- 
2.1.4

